services/wallet/api.go

```diff
@@ -567,18 +560,12 @@ func (api *API) CreateMultiTransaction(ctx context.Context, multiTransaction *Mu
 	return api.s.transactionManager.createMultiTransaction(ctx, multiTransaction, data, api.router.bridges, password)
 }
 
-func (api *API) IsCurrencyFiat(name string) bool {
-	code, _ := iso4217.ByName(strings.ToUpper(name))
-
-	return (code != 0)
+func (api *API) GetCachedCurrencyFormats() (currency.FormatPerSymbol, error) {
+	log.Debug("call to GetCachedCurrencyFormats")
+	return api.s.currency.GetCachedCurrencyFormats()
 }
 
-func (api *API) GetFiatCurrencyMinorUnit(name string) (int, error) {
-	code, minor := iso4217.ByName(strings.ToUpper(name))
-
-	if code == 0 {
-		return code, errors.New("Unknown currency: " + name)
-	}
-
-	return minor, nil
+func (api *API) FetchAllCurrencyFormats() (currency.FormatPerSymbol, error) {
+	log.Debug("call to FetchAllCurrencyFormats")
+	return api.s.currency.FetchAllCurrencyFormats()
```

Comment:
Why cannot we have only one endpoint?

Reply:
`GetCachedCurrencyFormats` is a quick, non-blocking operation so we use that during initialization, to ensure the client can properly represent currency amounts as strings even if there's no internet connection or whatever.

`FetchAllCurrencyFormats` might cause token prices to be fetched if they're not already cached. We don't know how long this will take, so we want to make it an async call.